File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/controllers/EmailController.java
package com.example.demo.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import com.example.demo.services.EmailService;

@RestController
@RequestMapping("/api/email")
@CrossOrigin(origins = "http://localhost:5173")
public class EmailController {

    @Autowired
    private EmailService emailService;

    @PostMapping("/send")
    public ResponseEntity<String> sendEmail(

            @RequestParam("email") String email,

            @RequestParam("pdf") MultipartFile pdf) {

        emailService.sendInvoice(email, pdf);

        return ResponseEntity.ok("Email Sent Successfully");
    }
}

File: EmailService/Controllers/EmailController.cs
using EmailService.Services;
using Microsoft.AspNetCore.Mvc;

namespace EmailService.Controllers
{
    [ApiController]
    [Route("api/email")]
    public class EmailController : ControllerBase
    {
        private readonly IEmailService _emailService;

        public EmailController(IEmailService emailService)
        {
            _emailService = emailService;
        }

        [HttpPost("send")]
        public async Task<IActionResult> SendEmail(
            [FromForm] string email,
            [FromForm] IFormFile pdf)
        {
            await _emailService.SendInvoiceAsync(email, pdf);

            return Ok("Email Sent Successfully");
        }
    }
}